src/svtk/svtk/pesr/sr_test.py

```diff
@@ -9,59 +9,61 @@
-        results = []
-        for coord in 'posA posB'.split():
-            result = self._test_coord(record, coord, called, background)
-            result['coord'] = coord
-            results.append(result)
-        results = pd.concat(results, ignore_index=True)
-
-        # Add test for sum of posA and posB
-        total = self._test_total(results)
-        results = pd.concat([results, total], ignore_index=True)
-
-        # Clean up columns
-        results['name'] = record.id
-        results['bg_frac'] = results.called / \
-            (results.background + results.called)
-        results['bg_frac'] = results.bg_frac.fillna(0)
-        cols = 'name coord pos log_pval called background bg_frac'.split()
-        return results[cols]
-
     def test_record(self, record, called, background):
         # Test SR support at all coordinates within window of start/end
         resultA = self._test_coord(
-            record, 'posA', record.chrom, called, background)
-        resultA['coord'] = 'posA'
+            record.pos, record.info['STRANDS'][0], record.chrom,
+            called, background, record.pos - self.window, record.pos + self.window)
 
         resultB = self._test_coord(
-            record, 'posB', record.info['CHR2'], called, background)
-        resultB['coord'] = 'posB'
-
-        if (record.chrom == record.info['CHR2']) and (int(resultA.pos) >= int(resultB.pos)):
-            rec = 1
-            while True:
-                result = self._test_coord_V2(
-                    record, 'posB', record.chrom, called, background, rec)
-                if int(result.pos) > int(resultA.pos.iloc[0]):
-                    result['coord'] = 'posB'
-                    resultB = result
-                    break
-                rec = rec + 1
+            record.stop, record.info['STRANDS'][1], record.info['CHR2'],
+            called, background, record.stop - self.window, record.stop + self.window)
+
+        posA = int(resultA.pos)
+        posB = int(resultB.pos)
+        log_pval_A = float(resultA.log_pval)
+        log_pval_B = float(resultB.log_pval)
+        if record.info['SVTYPE'] == 'INS':
+            if posA > posB + self.ins_window or posA < posB - self.ins_window:
+                # Invalid coordinates, need to re-optimize around the better coordinate within the insertion window size
+                if log_pval_A >= log_pval_B:
+                    # posA is better, so use posA as anchor and check for best posB in valid window
+                    resultB = self._test_coord(
+                        posA, record.info['STRANDS'][1], record.info['CHR2'],
+                        called, background, posA - self.ins_window, posA + self.ins_window)
+                else:
+                    # vice versa
+                    resultA = self._test_coord(
+                        posB, record.info['STRANDS'][0], record.chrom,
+                        called, background, posB - self.ins_window, posB + self.ins_window)
```

Comment:
Could this return a pair of coordinates that don't meet the requirements you are imposing? It seems that after being updated once they are not checked again

Reply:
It does allow for posB < posA, but that will get fixed later in `rewrite_SR_coords.py`. Is there another case you're thinking of?

Reply:
Yes, I guess it's just posA > posB. To clarify, why is that considered invalid on line 52 if it is fixed in rewrite_SR_coords.py?

Reply:
Ah, the difference here is the window size. In the first test we use `self.window` but here we only test over `self.ins_window`, which could be smaller.